src/lean_dojo/data_extraction/lean.py

```diff
@@ -43,27 +46,46 @@
     )
     GITHUB = Github()
 
-LEAN4_REPO = GITHUB.get_repo("leanprover/lean4")
+# LEAN4_REPO = GITHUB.get_repo("leanprover/lean4")
+LEAN4_REPO = None
 """The GitHub Repo for Lean 4 itself."""
 
-LEAN4_NIGHTLY_REPO = GITHUB.get_repo("leanprover/lean4-nightly")
+# LEAN4_NIGHTLY_REPO = GITHUB.get_repo("leanprover/lean4-nightly")
+LEAN4_NIGHTLY_REPO = None
 """The GitHub Repo for Lean 4 nightly releases."""
 
 _URL_REGEX = re.compile(r"(?P<url>.*?)/*")
 
 
-def normalize_url(url: str) -> str:
+def normalize_url(url: str, repo_type:str ='github') -> str:
+    if repo_type == 'local':
+        return os.path.abspath(url) # Convert to absolute path if local
     return _URL_REGEX.fullmatch(url)["url"]  # Remove trailing `/`.
 
 
 @cache
-def url_to_repo(url: str, num_retries: int = 2) -> Repository:
+def url_to_repo(url: str, num_retries: int = 2, repo_type: str = 'github') -> Repo:
+    """Convert a URL to a Repo object.
+    
+    Args:
+        url (str): The URL of the repository.
+        num_retries (int): Number of retries in case of failure.
+        repo_type (str): The type of the repository. Defaults to 'github'.
+
+    Returns:
+        Repo: A Git Repo object.
+    """
     url = normalize_url(url)
     backoff = 1
 
     while True:
         try:
-            return GITHUB.get_repo("/".join(url.split("/")[-2:]))
+            if repo_type == 'github':
+                return GITHUB.get_repo("/".join(url.split("/")[-2:]))
+            elif repo_type == 'local':
+                return Repo(url)
+            else:
+                return Repo.clone_from(url, '/tmp/repo_clone')  # We might need to change this with TMP_DIR
```

Comment:
The variable `TMP_DIR` is currently used only in the tracing process, which we can address later. non-github "remote" urls are simply cloned to `TMP_DIR`, similar to the local method. 

If these settings look good, I'll proceed with updating the tracing process. :)